### DISCUSSION TITLE
Ability to change the Document of a Controller in code

### DISCUSSION THREAD

**Discussion by WaterDLaw** [singerdmx/flutter-quill#979] [Feature Request]
One of the main reason to add this has to do with Image/File/Video uploads

1. If we upload images to a bucket we do not want to do that in the Image Embed hander, since there is always the option to remove the image again before the actual creation of the Document that uses the Quill editor.

2. An approach would be to change the delta with custom logic and later add it the the current controller and override the current delta. For example: 
controller.document = Document.fromJson(newDelta)

3. Currently the Document is final so we cannot change the document. All it needs is a set and getter for the document to the QuillController class.

Maybe i oversee certain things to achieve this in an other way, but I think this could be a nice feature. What would speak against to add something this simple?

**Comment by singerdmx** [comment]
Your questions are all solved already. Please check again
